File: net_tool/hosts_ports/scanport.py
import os
import socket
def ScanPort(address, start_port, end_port):
	available_ports=[]
	for port in range(start_port,end_port):
		s = socket.socket(socket.AF_INET,socket.SOCK_STREAM)
		try:
			s.connect((address,int(port)))
			s.shutdown(2)
			print("%d available!" %port)
		except:
			# print("%d is busy" %port)
			pass
	if not available_ports:
		print("Ports are all freaking busy")

if __name__ == '__main__':
	ScanPort("127.0.0.1",7776,7778)

File: net_tool/hosts_ports/scanport.rb
require 'socket'    
include Socket::Constants         
def scan(address, start_port, end_port)     
  threads = []     
  for port in start_port..end_port     
    threads << Thread.new(port) do |theport|     
      begin    
        socket = Socket.new(AF_INET, SOCK_STREAM, 0)     
        sockaddr = Socket.pack_sockaddr_in(theport, address)     
        socket.connect(sockaddr)     
        puts "Port:#{theport} is Opened\n"    
        socket.close     
      rescue    
      end    
    end    
  end    
  threads.each {|thr| thr.join}     
end
scan("127.0.0.1",8000,9080)#input here 